components/director/internal/domain/runtime_auth/repository.go

```diff
@@ -0,0 +1,133 @@
+package runtime_auth
+
+import (
+	"context"
+
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/kyma-incubator/compass/components/director/internal/repo"
+
+	"github.com/pkg/errors"
+)
+
+const tableName string = `"public"."runtime_auths"`
+
+var tableColumns = []string{"id", "tenant_id", "runtime_id", "api_def_id", "value"}
+
+//go:generate mockery -name=Converter -output=automock -outpkg=automock -case=underscore
+type Converter interface {
+	ToEntity(in model.RuntimeAuth) (Entity, error)
+	FromEntity(in Entity) (model.RuntimeAuth, error)
+}
+
+type pgRepository struct {
+	*repo.SingleGetter
+	*repo.Lister
+	*repo.Upserter
+	*repo.Deleter
+
+	conv Converter
+}
+
+func NewRepository(conv Converter) *pgRepository {
+	return &pgRepository{
+		SingleGetter: repo.NewSingleGetter(tableName, "tenant_id", tableColumns),
+		Lister:       repo.NewLister(tableName, "tenant_id", tableColumns),
+		Upserter:     repo.NewUpserter(tableName, tableColumns, []string{"tenant_id", "runtime_id", "api_def_id"}, []string{"value"}),
+		Deleter:      repo.NewDeleter(tableName, "tenant_id"),
+		conv:         conv,
+	}
+}
+
+func (r *pgRepository) Get(ctx context.Context, tenant string, apiID string, runtimeID string) (*model.RuntimeAuth, error) {
+	var ent Entity
+	if err := r.SingleGetter.Get(ctx, tenant, repo.Conditions{{Field: "runtime_id", Val: runtimeID}, {Field: "api_def_id", Val: apiID}}, &ent); err != nil {
+		return nil, err
+	}
+
+	runtimeAuthModel, err := r.conv.FromEntity(ent)
+	if err != nil {
+		return nil, errors.Wrap(err, "while creating runtime auth model from entity")
+	}
+
+	return &runtimeAuthModel, nil
+}
+
+func (r *pgRepository) GetOrDefault(ctx context.Context, tenant string, apiID string, runtimeID string) (*model.RuntimeAuth, error) {
+	var ent Entity
+
+	persist, err := persistence.FromCtx(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	stmt := `SELECT r.id AS runtime_id, r.tenant_id, ra.id, $2 AS api_def_id,
+	coalesce(ra.value, (SELECT default_auth FROM api_definitions WHERE api_definitions.id = $2)) AS value
```

Comment:
why there is `$2` in SELECT statement?

Reply:
Because we need API Definition ID to fetch default auth